pkg/applier/applier.go

```diff
@@ -187,25 +187,23 @@ func NewSupervisor(cs *ClientSet, scope declared.Scope, syncName string, reconci
 func (s *supervisor) UpdateStatusMode(ctx context.Context) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		u := newInventoryUnstructured(s.syncName, s.syncNamespace)
-		err := s.clientSet.Client.Get(ctx, client.ObjectKey{Name: u.GetName(), Namespace: u.GetNamespace()}, u)
+		key := client.ObjectKeyFromObject(u)
+		err := s.clientSet.Client.Get(ctx, key, u)
 		if err != nil {
 			// RG doesn't exist, it will be created by applier with appropriate status mode
 			if apierrors.IsNotFound(err) {
 				return nil
 			}
-			return err
+			return status.APIServerErrorf(err, "failed to get %s: %s", u.GetKind(), key)
```

Comment:
What is the motivation for wrapping this in `APIServerErrorf`?

Reply:
We do it almost everywhere else that uses the k8s api. It adds some special messaging for permission errors, and a KNV error code to help with SEO finding the known issues page.